### DISCUSSION TITLE
Should the build directory of a block plugin be committed to a repo?

### DISCUSSION THREAD

**Discussion by johnhooks** [WordPress/gutenberg#48227] [Q&A] [answered]
I'm learning how to create Gutenberg blocks and following the Create Block Tutorial. The `.gitignore` file created when using the `@wordpress/create-block` initializer doesn't excluded the `build` directory. So I kinda think it should be committed to the repo, but I'm coming from a frontside JavaScript background and committing build assets isn't standard practice.

Is the `./build` directory typically committed to a git repository?

**Comment by skorasaurus** [comment, marked as answer]
I'm not sure if there's quite any consensus yet and I wouldn't be too worried if you don't. 

Personally, I do commit it because if someone (including myself) has an issue with their build process locally and cannot build it, they can at least use the block without any extra actions. 

If you do include the build directory, just make sure that it is synced with your src directory. 

Some ([@ndiego](https://github.com/ndiego/block-visibility) , [@ryanwelcher](https://github.com/ryanwelcher/twitch/tree/trunk/plugins) do include the build directory; 

Others ( @richtabor (https://github.com/richtabor/share-on-twitter-block) [10up](https://github.com/10up/retro-winamp-block) don't. 


cc @ryanwelcher

**Reply by johnhooks** [reply]
Thanks for giving some examples, it really helps. I'll leave them out for now.
